components/src/core/components/TableFilter/TableFilter.vue

```diff
@@ -44,24 +46,69 @@ export default defineComponent({
       type: String,
       required: true,
     },
+    itemsSelected: {
+      type: Boolean,
+      default: false,
+    },
   },
 
-  setup() {
+  setup(props) {
     const responsiveState = useResponsive();
     const isActive = ref(true);
+    const isSpilled = ref(true);
 
     const toggleFilters = () => {
       isActive.value = !isActive.value;
     };
 
+    const updateIsSpilled = () => {
+      nextTick(function() {
+        const titleWidth = document.getElementsByClassName(
+          'oxd-table-filter-header-title',
+        )[0]
+          ? document.getElementsByClassName('oxd-table-filter-header-title')[0]
+              .clientWidth
+          : 0;
+        const actionButtonWidth = document.getElementsByClassName(
+          'oxd-table-filter-header-bulk-actions',
+        )[0]
+          ? document.getElementsByClassName(
+              'oxd-table-filter-header-bulk-actions',
+            )[0].clientWidth
+          : 0;
+        const fullWidth = document.getElementsByClassName(
+          'oxd-table-filter-header',
+        )[0]
+          ? document.getElementsByClassName('oxd-table-filter-header')[0]
+              .clientWidth
+          : 0;
+        const filterWidth = document.getElementsByClassName(
+          'oxd-table-filter-header-options',
+        )[0]
+          ? document.getElementsByClassName(
+              'oxd-table-filter-header-options',
+            )[0].clientWidth
+          : 0;
+        isSpilled.value =
+          titleWidth + actionButtonWidth + 16 > fullWidth - filterWidth;
```

Comment:
This 16 is coming from the padding of the outer element. But this is would be error-prone. Try removing padding before getting the width. If that is not possible at least CSS should be denoted in a comment.